File: elgjava/sl/crypto/elgamal/keys/Key.java
package sl.crypto.elgamal.keys;

import sl.crypto.elgamal.parameter.Parameter;

/**
 * Apstrakte Klasse welche ein Key Objekt beschreibt
 * 
 * @author Matthias Koch
 * @version 0.5  
 */
public abstract class Key
{
	private Parameter params;
	boolean privateKey;
	/**
	 * 
	 * @return boolean - gibt an ob es sich um einen privaten Schlüssel handelt
	 */
	public boolean isPrivate()
	{
		return privateKey;
	}
	/**
	 * 
	 * @param isPrivate
	 * @param params
	 */
	protected Key(boolean isPrivate, Parameter params)
	{
		this.privateKey = isPrivate;
		this.params = params;
	}
	/**
	 * 
	 * @return Parameter gibt die Paramter des Schlüssels zurück 
	 */
	public Parameter getParameters()
	{
		return params;
	}
	/**
	 * 
	 */
	public boolean equals(Object obj)
	{
		if (!(obj instanceof Key))
		{
			return false;
		}
		Key dhKey = (Key) obj;
		return (params != null && !params.equals(dhKey.getParameters()));
	}
}

File: elgcsharp/Key.cs
namespace sl.crypto.elgamal.keys
{
	using sl.crypto.elgamal.parameter;
	/**
	 * <summary>
 	 * Autor: Matthias Koch
 	 * Version: 0.5
 	 * 
 	 * Apstrakte Klasse welche ein Key Objekt beschreibt
 	 * </summary> 
 	 */
	public abstract class Key
	{
		private Parameter parameter;
		//erlaube lesenden Zugriff
		public Parameter Parameter
		{
			get{return parameter;}
		}
		private bool privateKey;
		//erlaube lesenden Zugriff
		public bool isPrivateKey
		{
			get{return privateKey;}
		}
		/**
		 * 
		 * <param name="isPrivate"/>
		 * <param name="params"/>
		 */
		protected Key(bool isPrivate, Parameter parameter)
		{
			this.privateKey = isPrivate;
			this.parameter = parameter;
		}
		/**
		 * 
		 */
		public new bool Equals(object obj)
		{
			if (!(obj is Key))
			{
				return false;
			}
			Key dhKey = (Key) obj;
			return (parameter != null && !parameter.Equals(dhKey.Parameter));
		}
	}
}